src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -479,16 +411,22 @@ UprobesWithStackPerfEvent ConsumeUprobeWithStackPerfEvent(PerfEventRingBuffer* r
 
 GenericTracepointPerfEvent ConsumeGenericTracepointPerfEvent(PerfEventRingBuffer* ring_buffer,
                                                              const perf_event_header& header) {
-  perf_event_raw_sample_fixed ring_buffer_record;
-  ring_buffer->ReadRawAtOffset(&ring_buffer_record, 0, sizeof(perf_event_raw_sample_fixed));
+  // The flags here are in sync with generic_event_attr in PerfEventOpen.
+  // TODO(b/242020362): use the same perf_event_attr object from generic_event_attr
+  const perf_event_attr flags{
+      .sample_type = SAMPLE_TYPE_TID_TIME_STREAMID_CPU,
```

Comment:
Wait, is that correct? Shouldn't we always set the SAMPLE_RAW?

Reply:
Yes, it's correct. This function only reads the fields that are common between all tracepoints. It doesn't read tracepoint specific fields. Reading SAMPLE_RAW will differ based on what type of tracepoint you're trying to read.

Reply:
Thanks for checking. So I guess the "generic trace points" are used for the trace point track, were we do not show the actual tracepoint data, but only a timestamp.